--- a/hw/display/virtio-gpu-base.c
+++ b/hw/display/virtio-gpu-base.c
@@ -233,6 +233,7 @@
     }
 
     g->virtio_config.num_scanouts = cpu_to_le32(g->conf.max_outputs);
+    g->virtio_config.blob_alignment = cpu_to_le32(qemu_real_host_page_size());
     virtio_init(VIRTIO_DEVICE(g), VIRTIO_ID_GPU,
                 sizeof(struct virtio_gpu_config));
 
@@ -275,6 +276,9 @@
     }
     if (virtio_gpu_resource_uuid_enabled(g->conf)) {
         features |= (1 << VIRTIO_GPU_F_RESOURCE_UUID);
+    }
+    if (virtio_gpu_blob_enabled(g->conf)) {
+        features |= (1 << VIRTIO_GPU_F_BLOB_ALIGNMENT);
     }
 
     return features;
